dbsc/workflow/rules/mrtpipelines.smk

```diff
@@ -1,571 +1,600 @@
 from os.path import join
 
-include: 'zona_bb_subcortex.smk'
-include: 'freesurfer.smk'
 
-# DWI Processing
-rule nii_to_mif:
+include: "zona_bb_subcortex.smk"
+include: "freesurfer.smk"
+
+
+# Directories
+mrtrix_dir = join(config["derivatives"], "mrtrix")
+
+# Paramaters
+responsemean_flag = config.get("responsemean_dir", None)
+shells = config.get("shells", "")
+lmaxes = config.get("lmax", "")
+
+# Mrtrix3 citation (additional citations are included per rule as necessary):
+# Tournier, J.-D.; Smith, R. E.; Raffelt, D.; Tabbara, R.; Dhollander, T.; Pietsch, M.; Christiaens, D.; Jeurissen, B.; Yeh, C.-H. & Connelly, A. MRtrix3: A fast, flexible and open software framework for medical image processing and visualisation. NeuroImage, 2019, 202, 116137
+
+
+# ------------ MRTRIX PREPROC BEGIN ----------#
+rule nii2mif:
     input:
-        dwi=bids(
-            root=join(config["bids_dir"], 'derivatives/prepdwi')
-            datatype='dwi',
-            suffix='dwi.nii.gz',
-            space='T1w',
-            **config['subj_wildcards'],
-        )
-        bval=bids(
-            root=join(config["bids_dir"], 'derivatives/prepdwi')
-            datatype='dwi',
-            suffix='dwi.bval',
-            space='T1w',
-            **config['subj_wildcards'],
-        )
-        bvec=bids(
-            root=join(config["bids_dir"], 'derivatives/prepdwi')
-            datatype='dwi',
-            suffix='dwi.bvec',
-            space='T1w',
-            **config['subj_wildcards'],
-        )
-        mask=bids(
-            root=join(config["bids_dir"], 'derivatives/prepdwi')
-            datatype='dwi',
-            suffix='brainmask.nii.gz',
-            space='T1w',
-            **config['subj_wildcards'],
-        )
-    params:
-        threads=workflow.cores,
+        dwi=inputs["dwi"].input_path,
+        bval=lambda wildcards: re.sub(".nii.gz", ".bval", inputs["dwi"].input_path),
```

Comment:
I don't think these need to be input functions (i.e. I think you could get rid of `lambda wildcards: `

Reply:
Fair - this was copied from another workflow. I just realized I don't think `re` is imported here either. I guess 2 things to fix:

1. `lambda wildcards:`
2. `import re`